vision.md

```diff
@@ -0,0 +1,41 @@
+
+1. Auto-log any error appearing in the nightlies, even when there is no one around. This could serve future investigations.
+2. Any person can be assigned to Buildfarmer Payload in case of necessity, this implies:
+   * Basic duties should be easy to automate, even scripted.
+   * Everyone assigned to Buildfarmer Payload must log/communicate internally any actions completed in the name of the role, so other folks can make progress on top of previous actions.
+   * There will be a public log with investigations related to the Buildfarmer Payload duties.
+   * Actions related to Buildfarmer Payload will be differentiated from actions related to the infra team, it is not resposability of the role to investigate infra issues, instead those will be escalated.
+3. Core Team and community developers should be able to check if an existent error in his build it’s also happening in the nightlies CI.
+4. Consecutive test failures priorities policy: Test errors occuring consecutively will be prioritized in terms of order of investigation, in the case of Gazebo jobs, the failing build will be retriggered to confirm if the error occurs a third time in a row. Errors happening three times in a row will be reported immediatly to the development teams in case the buildfarmer investigation wasn't enough to solve it. Errors that have appeared at least three times every 2 weeks, will also be prioritized. Buildfarmer Payload scripts will prioritize accordingly new investigations based on this.
+5. Flaky tests will be categorized based on how often they appear in a given month in a unstable builds.
+   * **~7% or errors that happened only once** Will not be reported.
+   * **7% - 20%** Will be reported and announced in the weekly meetings. This category will be considered as known issues, as we won’t aim to disable those tests.
+   * **20% - 100%** Will be reported, investigated internally, and eventually assigned someone from the development teams. Flaky tests falling in this category will be expected to be disabled 3 weeks after being assigned a developer to take a look or after being reported.
+6. The buildfarmer team will keep track of all the tests disabled internally and generate a report monthly. That way any developer can track those and aim to fix the underlying issues.
+7. As a new error gets investigated, make sure to close the reported issues/investigations ONLY after the issue is addressed in all combinations of job, platform, distribution where it was reported. Group errors in one a single issue if there's evidence they share an underlying reason (appeared after the same date, they occur in the same jobs, etc).
+8. Reports to generate monthly:
+   * Buildfarm average greenness per distro / platform (TBD)
+   * Disabled tests: Those can be obtained using the jenkins API, will be saved to our internal database.
```

Comment:
Can you verify that it is indeed possible to obtain this using the Jenkins API? For example, https://build.osrfoundation.org/view/gz-harmonic/job/gz_sim-8-win/24/testReport/ shows only 3 tests skipped, which I assume is what the API would tell you, but we have a lot more tests disabled on windows with the `DISABLED_` gtest prefix (see https://google.github.io/googletest/advanced.html#temporarily-disabling-tests). The ones that show up as skipped actually run but immediately call `GTEST_SKIP`.

Reply:
I don't think that Jenkins can show disabled tests (versus skipped test) since the disabled tests produce no output in the xunit/junit xml which is what Jenkins uses to assemble its test reporting. We would have to use a different tool or see if we can tweak ctest/gtest to emit a SKIP result for disabled tests.

Reply:
Yes, Steven! is right, I will modify this text to reflect that we will track in the database skipped tests as tracked by Jenkins.

We don't have a way to track disabled tests by now automatically, ideally the dev teams will let us know whenever tests are being disabled so we can save them in our database and included it as part of the monthly report.